crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2469,6 +2518,60 @@ impl std::fmt::Display for DynamicType {
     }
 }
 
+bitflags! {
+    /// Type qualifiers that appear in an annotation expression.
+    #[derive(Copy, Clone, Debug, Eq, PartialEq)]
+    pub(crate) struct TypeQualifiers: u8 {
+        /// `typing.ClassVar`
+        const CLASS_VAR = 1 << 0;
+        /// `typing.Final`
+        const FINAL     = 1 << 1;
+    }
+}
+
+/// When inferring the type of an annotation expression, we can also encounter type qualifiers
+/// such as `ClassVar` or `Final`. These do not affect the inferred type itself, but rather
+/// control how a particular symbol can be accessed or modified. This struct holds a type and
+/// a set of type qualifiers.
+///
+/// Example: `Annotated[ClassVar[tuple[int]], "metadata"]` would have type `tuple[int]` and
+/// qualifiers `CLASS_VAR | FINAL`.
```

Comment:
Would it? I don't see `Final` in this example ;)

Reply:
Thanks. It was there before, but I removed it and forgot to update the qualifier expression.